app/src/rgb_underglow.c

```diff
@@ -434,4 +440,67 @@ int zmk_rgb_underglow_change_spd(int direction) {
     return zmk_rgb_underglow_save_state();
 }
 
+#if IS_ENABLED(CONFIG_ZMK_RGB_UNDERGLOW_AUTO_OFF_IDLE)
+static bool auto_off_idle_prev_state = false;
+#endif
+
+#if IS_ENABLED(CONFIG_ZMK_RGB_UNDERGLOW_AUTO_OFF_USB)
+static bool auto_off_usb_prev_state = false;
+#endif
+
```

Comment:
I'm not sure we actually need dedicated *different* variables for each condition, instead of one general `auto_off_prev_state` boolean.

Reply:
We cannot use a single variable. Consider the following situation when both `AUTO_OFF_IDLE` and `AUTO_OFF_USB` are enabled.

* initially `state=on`
* unplug USB: `state=off`, `prev_state=on`
* idle state: nothing happens
* back from idle state
  expected behaviour: nothing happens
  actual behaviour: RGB turns on since `prev_state=on`

Reply:
Yeah I think separating the two is better because of how the two states can conflict and interact with each other. I'm happy to implement it and test it but I'm pretty sure @bortoz is correct in that unexpected behaviour can happen

Reply:
You should still be able to avoid the duplication then, by making the function take a pointer to the state variable you should use for comparison and update. That helps bring the code size down, while avoiding the edge cases you note. Thanks!